handbook/engineering/hiring/software-engineer-frontend.md

```diff
@@ -1,19 +1,34 @@
 # Software Engineer - Frontend
 
-We are looking for frontend engineers who know how to build intuitive user experiences and APIs to make the power of [Universal Code Search](https://about.sourcegraph.com/product) accessible to everyone. You will have a lot of ownership to solve tough technical and UX problems in key areas of our web application and browser extensions (e.g. code search, code navigation, campaigns).
+We are looking for frontend engineers who know how to build intuitive user experiences to make the power of [Universal Code Search](https://about.sourcegraph.com/product) accessible to everyone. You will have a lot of ownership to solve tough technical and UX problems in key areas of our web application and browser extensions (e.g. code search, code navigation, campaigns).
+
+## What you will do
+
+- Work closely with a dedicated designer and PM to implement high-quality UIs.
+- Actively participate in the planning of the features we're going to build, e.g. by writing and reviewing RFCs.
+- Build UIs you will love to use yourself as a developer.
+- Ensure UI/UX consistency in our app.
+- Identify potential for abstraction and build components that can be reused, even by other teams.
+- Improve the code patterns in our frontend.
+- Hold up accessibility standards.
 
 ## Qualifications
 
-- Excellent knowledge of TypeScript.
-- Solid understanding of JavaScript's concurrency model and event loop.
-- Skilled at building and testing (e.g., unit testing, automated end-to-end testing) React single page applications and HTTP APIs (GraphQL and Go preferred).
+- Experience in writing [SPAs](https://en.wikipedia.org/wiki/Single-page_application) using React or similar component-based frontend frameworks.
+- Familiarity with TypeScript or a different typed programming language.
+- Knowledge of CSS.
+- Knowledge of semantic HTML.
```

Comment:
I think it is worth grouping these and focusing on the applied aspect (using these skills to achieve an outcome, vs just knowing them).
```suggestion
- Skilled at writing HTML and CSS to implement visual designs.
```

Reply:
What I want to make clear here is that we're not looking for someone who just "makes it work" and look like the design after doing a bunch of absolute pixel positioning, but someone who has deep enough knowledge to establish patterns and abstractions that can be followed/used by others. I think the suggestion would be a step back in that regard, because it removes the keyword "semantic" and like you said puts the focus on the user-observable outcome.
I do see though how the previous points are so generic that everyone would just think ☑️  and read over them. Do you have an idea for something solving both concerns?

Reply:
> knowledge to establish patterns and abstractions that can be followed/used by others

Say that then!

```suggestion
- Skilled at writing clean and reusable HTML and CSS to implement visual designs.
```